src/parser/Language/Haskell/Tools/Parser/RemoveWildCards.hs

```diff
@@ -0,0 +1,127 @@
+                        then foldl' (\acc x -> acc ++ [addAsPat x Nothing]) [] args
+                        else snd $ foldl' (\(c,acc) x -> (c+1,acc ++ [addAsPat x (Just c)])) (0,[]) args
+   in
+    mkMatchLhs' name $ modifiedArgs
+  where
+    addAsPat :: Ann UPattern (Dom GhcPs) SrcTemplateStage -> Maybe Int -> Ann UPattern (Dom GhcPs) SrcTemplateStage
+    addAsPat expr@(Ann _ (UParenPat (Ann _ (URecPat patternName _)))) count =
+        let (firstLetter, newName) = fromMaybe ('a', []) $ uncons $ getFieldName patternName
+        in mkAsPat' (mkName' ([toLower firstLetter] ++ newName ++ (fromMaybe "" $ show <$> count))) expr
+    addAsPat expr@(Ann _ (URecPat patternName _)) count =
+        let (firstLetter, newName) = fromMaybe ('a', []) $ uncons $ getFieldName patternName
+        in mkAsPat' (mkName' ([toLower firstLetter] ++ newName ++ (fromMaybe "" $ show <$>  count))) expr
+    addAsPat expr _ = expr
+changeAsPat expr = pure expr
+
+traverseOverUValBind :: Ann UDecl (Dom GhcPs) SrcTemplateStage -> IO (Ann UDecl (Dom GhcPs) SrcTemplateStage)
+traverseOverUValBind expr@(Ann _ (UValueBinding (FunctionBind' ex))) = do
+    !wildCardList <- mapMaybeM (wildCardsRefactor) (ex ^? biplateRef)
+    print (HM.unions wildCardList)
+    let newList = lookUpAndConcat (HM.unions wildCardList)
+    print newList
+    newAST <- (!~) (biplateRef) (removeWildPats) (expr)
+    newAST' <- (!~) (biplateRef) (changeWildCards newList) (newAST)
+    -- pure expr
+    pure newAST'
+  where
+    lookUpAndConcat :: (HM.HashMap String (String, String)) -> (HM.HashMap String String)
+    lookUpAndConcat hashMap = HM.foldlWithKey (\acc k v -> HM.insert k (fromMaybe "" $ getValue k v hashMap) acc) HM.empty hashMap
+    getValue k v hashMap =
+        case HM.lookup k hashMap of
+          Nothing -> Nothing
+          Just (a,b) -> Just $ maybe (a ++ "." ++ b) (\x -> x ++ "." ++ b) $ getValue a v hashMap
+traverseOverUValBind expr = pure expr
+
+getFieldName :: Ann UName (Dom GhcPs) SrcTemplateStage -> String
+getFieldName expr@(Ann _ (UNormalName (Ann _ (UQualifiedName _ (Ann _ (UNamePart ex)))))) = ex
+getFieldName expr = ""
+
+changeWildCards :: HM.HashMap String (String) -> Ann UName (Dom GhcPs) SrcTemplateStage -> IO (Ann UName (Dom GhcPs) SrcTemplateStage)
+changeWildCards list expr@(Ann _ (UNormalName (Ann _ (UQualifiedName _ (Ann _ (UNamePart ex)))))) = 
+  pure
+   $ case HM.lookup ex list of
+       Nothing -> expr
+       Just val -> mkName' val
+changeWildCards _ expr = pure expr
+
+getNameFromPattern :: Ann UPatternField (Dom GhcPs) SrcTemplateStage -> (String, String)
+getNameFromPattern expr@(Ann _ (UNormalFieldPattern fieldName (Ann _ (UVarPat (Ann _ (UNormalName (Ann _ (UQualifiedName _ (Ann _ (UNamePart ex)))))))))) = (getFieldName fieldName, ex)
+getNameFromPattern expr@(Ann _ (UNormalFieldPattern fieldName ex@(Ann _ (UAsPat patternName _)))) = (getFieldName fieldName,getFieldName patternName)
+getNameFromPattern expr = ("", "")
```

Comment:
empty is fine? or better to make it Maybe ?

Reply:
empty is fine because its just string concatination